### ISSUE TITLE
GTM installation

### ISSUE THREAD

**Issue by chensinging** [twreporter/twreporter-react#1798]
請將以下程式碼複製並貼入您網站的每個網頁中。

請將這段程式碼放在網頁最上方的 `<head>` 中：

```
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-PRMXBBN');</script>
<!-- End Google Tag Manager -->
```

此外，請將這段程式碼複製到 `<body>` 開頭代碼正後方：

```
 <body> 
<!-- Google Tag Manager (noscript) -->
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-PRMXBBN"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<!-- End Google Tag Manager (noscript) -->
```

**Comment by chensinging** [comment]
# Scroll Depth with GTM

This component will measure specific events to better identify users that reached specific article completion milestones.

## Milestone 25%

Add the following Javascript code snippet to this event. Change the <article title> variable to the actual article title present where the measuring event is fired:

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'article-milestone',
'article_title': '<article title>',
'scroll_measure_percent': '25%',
'scroll_measure_value': 25
});`

## Milestone 50%

Add the following Javascript code snippet to this event. Change the <article title> variable to the actual article title present where the measuring event is fired:

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'article-milestone',
'article_title': '<article title>',
'scroll_measure_percent': '50%',
'scroll_measure_value': 50
});`

## Milestone 75%

Add the following Javascript code snippet to this event. Change the <article title> variable to the actual article title present where the measuring event is fired:

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'article-milestone',
'article_title': '<article title>',
'scroll_measure_percent': '75%',
'scroll_measure_value': 75
});`

## Milestone 100%

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'article-milestone',
'article_title': '<article title>',
'scroll_measure_percent': '100%',
'scroll_measure_value': 100
});`

**Comment by chensinging** [comment]
# Social Sharing

This component will measure specific events to better identify users who are sharing articles on specific social platforms.

## Facebook

Add the following Javascript code snippet to your code block:

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'social-share',
'social_platform': 'facebook'
});`

## Messenger

Add the following Javascript code snippet to your code block:

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'social-share',
'social_platform': 'messenger'
});`

## Twitter

Add the following Javascript code snippet to your code block:

`window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'social-share',
'social_platform': 'twitter'
});`

**Comment by chensinging** [comment]
# Push Notification Subscription

This component will measure specific events to better identify users who clicked on a push notification subscription offer.

Add the following Javascript code snippet to the point in your code block where the user has successfully subscribed to push notifications. **Change the "article title" variable to the actual article title present where the measuring event is fired:**

```
window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'push-notification-click',
'article_title': '<article title>'
});
```

# Add to Homescreen

Add the following Javascript code snippet to the point in your code block where the user has successfully added the Progressive Web App to their homescreen. Use the built-in GTM 'Page URL" for the event label:

```
window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
'event' : 'add-to-homescreen'
});
```

**Comment by babygoat** [comment]
@taylrj  Does [about-us index page](https://github.com/twreporter/twreporter-react/blob/master/src/components/about-us/index.js) and [index page](https://github.com/twreporter/twreporter-react/blob/master/src/containers/Home.js) support server side rendering? (i.e. window might not exist in these components?)

**Comment by taylrj** [comment]
about-us index page: no
index page: yes

**Comment by taylrj** [comment]
Resolved and landed in [v4.6.11](https://github.com/twreporter/twreporter-react/releases/tag/v4.6.11).
